traitsui/tree_node.py

```diff
@@ -1889,11 +1892,11 @@ def tno_when_label_changed(self, node, listener, remove):
         """
         label = node.label
         if label[:1] != '=':
-            memo = ('label', label, object, listener)
```

Comment:
So it was grabbing the `object` builtin?

Reply:
Yep.  This is a dumb error on my part:

- I assumed `TreeNodeObject` subclassed `TreeNode`
- I then copy-pasted and didn't pay attention to the context fully.

Reply:
I guess the icing on the cake was Mayavi's use of `TreeNodeObject` as the base of it's `Base` class?

Reply:
Yep, whereas I almost never use `TreeNodeObject`.

I've added a regression test.